packages/runtime/src/state/modules/read-only-documents.ts

```diff
@@ -1,6 +1,15 @@
+import { ValueJSON } from '../../controls'
 import { Action, ActionTypes } from '../actions'
 
-export type Data = undefined | null | boolean | number | string | Data[] | { [key: string]: Data }
+export type Data =
+  | undefined
+  | null
+  | boolean
+  | number
+  | string
+  | Data[]
+  | { [key: string]: Data }
+  | ValueJSON
```

Comment:
We should _not_ do this. `ValueJSON` should already extend `Data` given how its defined. I've seen some weird issues with interfaces not matching with the type and there was a utility function I made to work around it called `IndexSignatureHack`, maybe that can help again? If not, we can use `// @ts-expect-error` comments in specific places.